subprojects/ide-native/src/main/java/org/gradle/ide/visualstudio/internal/VisualStudioTargetBinary.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.ide.visualstudio.internal;
+
+import org.gradle.api.file.FileCollection;
+
+import java.io.File;
+import java.util.List;
+
+/**
+ * A model-agnostic adapter for binary information
+ */
+public interface VisualStudioTargetBinary {
+    /**
+     * Returns the project path for this binary
+     */
+    String getProjectPath();
+
+    /**
+     * Returns the name of the component associated with this binary
+     */
+    String getComponentName();
```

Comment:
Maybe just collapse this and project path into a single name?

Reply:
At the moment, I think we still have to reference them separately, so we probably want to store them separately for now.  Some of that goes away when we move to a single solution, so we might be able to wipe this out then.